test/core/util/test_config.cc

```diff
@@ -388,27 +388,6 @@ TestEnvironment::TestEnvironment(int argc, char** argv) {
 }
 
 TestEnvironment::~TestEnvironment() {
```

Comment:
I like to remove this with this approach but it's not possible because the final shutdown is not called by the main thread for some tests, which leads an async shutdown even with this PR.

Reply:
Restored the previous version.